src/parser/tree.ts

```diff
@@ -97,30 +99,35 @@ export function doTree(
 				i18nFunctions[functionName as keyof typeof i18nFunctions]
 
 			const children = raw.children.slice(1, -1)
-			let translationKeyIndex = 0;
+			let translationKeyIndex = 0
 
 			// Get the translation from the arguments (the quoted strings)
 			for (const child of children) {
-				let node = child;
+				let node = child
+				let nodeValue = node.text
 
 				// unwrap the argument node, which is used in PHP.
-				if (child.type === "argument") {
-					if (child.children.length === 0) continue;
-					node = child.children[0];
+				if (child.type === 'argument') {
+					if (child.children.length === 0) continue
+					node = child.children[0]
 				}
 
-				// skip if it's not a string
-				if (node.type !== "string") continue
+				if (node?.type === ',') {
+					// skip the comma between arguments
+					continue
+				} else if (stringType.includes(node?.type)) {
+					// unquote the strings
+					nodeValue = nodeValue.slice(1, -1)
+				}
```

Comment:
@erikyo why do an `else if` here? If this isn't true, shouldn't we not continue?
```suggestion
				}
				
				if (stringType.includes(node?.type)) {
					// unquote the strings
					nodeValue = nodeValue.slice(1, -1)
				} else {
				  // unexpected node type
				  continue
				}
```

Reply:
Or if we hit and unexpected node type, perhaps we shouldn't complete the translation, and should flag it to warn about improper use of the function.

Reply:
tested locally but seems that this change can introduce a new glitch, problably because that values aren't a "string" type
![image](https://github.com/wp-blocks/make-pot/assets/8550908/3cc98ed3-b416-4ccf-9d71-6954272185e6)

i'll look into to find the correct names

Reply:
the missing types (using a really wide base of plugins and themes so i'm quite sure that i'm covering all the possible cases):

- shell_command_expression -> `This is a ${bug}` (with the "`" quotes)
- function_call_expression -> count( %s ) 
- variable_name -> $variable
- name -> it's a js variable


@johnhooks what do you suggest with those types? aren't supported but as you said we should provide a fallback. why not use that replacing the continue with a console.warn?

Reply:
@erikyo i would agree that if an unknown type is encountered, it should be logged, and not translation not included in the output.

Reply:
ok then... you are right! i'll add back the `continue`, but a will also keep the `console.warn` in order to inform the user about the reason of the missing translation